### PR TITLE
Hooks Exercise 1 + Extra Credit

### PR DESCRIPTION
**Pull request by dangrous**
I have two questions about his solution that I'll thread off of the Slack post. Also I had my initial solution which worked but I see why this is better, in the commit chain.

### CODE REVIEW

File: src/exercise/01.js [hunk 1]
```diff
@@ -3,27 +3,25 @@
...
-    // 🐨 update the name here based on event.target.value
+    setName(event.target.value)
   }
-
   return (
     <div>
       <form>
         <label htmlFor="name">Name: </label>
-        <input onChange={handleChange} id="name" />
+        <input value={name} onChange={handleChange} id="name" />
```
Comment (benpbolton) [inline src/exercise/01.js:16]: `value` should not be needed here; interesting twist....

File: src/exercise/01.js [hunk 2]
```diff
@@ -3,27 +3,25 @@
 
 import * as React from 'react'
 
-function Greeting() {
-  // 💣 delete this variable declaration and replace it with a React.useState call
-  const name = ''
+function Greeting({initialName = ''}) {
```
Comment (benpbolton) [inline src/exercise/01.js]: I like how you navigated around the props... it's kindof clever... BUT... it's probably worth explicitly still using `props` here and below so you get the hang of callin' 'em props... so something like:

``
```suggestion
function Greeting(props) {
```

with a later....

File: src/exercise/01.js [hunk 3]
```diff
@@ -3,27 +3,25 @@
...
+  const [name,setName] = React.useState(initialName)
```
Comment (benpbolton) [inline src/exercise/01.js]:
```suggestion
  const [name, setName] = React.useState(props.initialName);
```

**Review by benpbolton — APPROVED** [review]
Left some propsy thoughts... and no need for value.. I don't think!

File: src/exercise/01.js (hunk 1 above)
Comment (dangrous) [reply src/exercise/01.js:16]: Yeah he had that in his to prepopulate the field... seemed nice so I kept it haha

File: src/exercise/01.js (hunk 3 above)
Comment (dangrous) [reply src/exercise/01.js]: yeah see i did that first but then his version didn't have ittttttttt ah well.

**Pull request merged**